bridge-token/src/lib.rs

```diff
@@ -65,38 +73,22 @@ impl BridgeToken {
     }
 }
 
-#[near_bindgen]
-impl FungibleToken for BridgeToken {
-    #[payable]
-    fn inc_allowance(&mut self, escrow_account_id: String, amount: U128) {
-        self.token.inc_allowance(escrow_account_id, amount.into());
-    }
-
-    #[payable]
-    fn dec_allowance(&mut self, escrow_account_id: String, amount: U128) {
-        self.token.dec_allowance(escrow_account_id, amount.into());
-    }
+near_contract_standards::impl_fungible_token_core!(BridgeToken, token);
+near_contract_standards::impl_fungible_token_ar!(BridgeToken, token);
 
-    #[payable]
-    fn transfer_from(&mut self, owner_id: String, new_owner_id: String, amount: U128) {
-        self.token
-            .transfer_from(owner_id, new_owner_id, amount.into());
-    }
-
-    #[payable]
-    fn transfer(&mut self, new_owner_id: String, amount: U128) {
-        self.token.transfer(new_owner_id, amount.into());
-    }
-
-    fn get_total_supply(&self) -> U128 {
-        self.token.get_total_supply().into()
-    }
-
-    fn get_balance(&self, owner_id: String) -> U128 {
-        self.token.get_balance(owner_id).into()
-    }
-
-    fn get_allowance(&self, owner_id: String, escrow_account_id: String) -> U128 {
-        self.token.get_allowance(owner_id, escrow_account_id).into()
+#[near_bindgen]
+impl FungibleTokenMetadataProvider for BridgeToken {
+    fn ft_metadata(&self) -> FungibleTokenMetadata {
+        FungibleTokenMetadata {
+            spec: FT_METADATA_SPEC.to_string(),
+            name: self.name.clone(),
+            symbol: self.symbol.clone(),
+            icon: Some(
+                "https://near.org/wp-content/themes/near-19/assets/img/brand-icon.png".to_string(),
+            ),
+            reference: self.reference.clone(),
+            reference_hash: self.reference_hash.clone(),
+            decimals: 24,
```

Comment:
Decimals data may be incorrect, let's use here some odd number like 0.

Reply:
Rather than using 0, added decimal field to the contract itself so we can update it.